src/zhinst/toolkit/nodetree/helper.py

```diff
@@ -178,3 +178,18 @@ def to_dict(self) -> t.Dict[str, t.Any]:
         After conversion, :class:`Node` objects cannot be used to get items.
         """
         return self._result
+
+
+def not_callable_in_transactions(func: t.Callable[['Node', ...], None]):
+    """Wrapper to prevent certain functions from being used within a transaction.
+
+    Certain utils functions which that both get and set values would not work like
+    expected in a transaction. This wrapper prevents misuse by throwing an error
+    in such cases.
+    """
+    def f(self: 'Node', *args, **kwargs):
```

Comment:
the name `f` is not very descriptive. I would use something more readable e.g. ` wrapper`